File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    
    for c in commands:
        i,j,k = c
        #1. i부터 j까지 자름
        arr = array[i-1:j]
        #2. 정렬함
        arr.sort()
        #3. k번째 수를 answer에 담음
        answer.append(arr[k-1])
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        for(int x=0; x<commands.length; x++){
            int i = commands[x][0];
            int j = commands[x][1];
            int k = commands[x][2];
            int[] arr = new int[j-i+1];
            System.arraycopy(array, i-1, arr, 0, j-i+1);
            
            Arrays.sort(arr);
            answer[x] = arr[k-1];
        }
        return answer;
    }
}